estate/models/estate_property_offer.py

```diff
@@ -57,3 +57,8 @@ def action_refuse_offer(self):
         self.property_id.write({"selling_price": 0})
         self.write({"status": "refused"})
         return True
+
+    @api.model
+    def create(self, vals):
+        self.env["estate.property"].browse(vals["property_id"]).check_offer_price(vals["price"])
+        return super().create(vals)
```

Comment:
You can do here the validation of the price and then set the status in vals.
```suggestion
        property = self.env["estate.property"].browse(vals["property_id"])
        max_price = max(property.offer_ids.mapped("price"))
        if max_price and float_compare(vals["price"], max_price, precision_digits=2) == -1:
            raise UserError("The offer price cannot be lower than %.2f" % max_price)
        vals.update({"status":  "offer_received"})
        return super().create(vals)
```

Reply:
I changed "property" to "actual_property" in the next PR (my mistake) because the pre-commit give me this error "shadows built in name" and I didn't notice.